### PR TITLE
Add string and path concatenation operators

### LINKED ISSUE #2: What inplace operator should we use for path concatenation?
The `join` function currently is there, and I see no reason to change anything about it.

But it would be nice to also have an inplace operator that does the same trick. I'll list some candidates as separate comments, so that folks can like/dislike them individually. Any reasons, explanations, new suggestions etc. would also be most welcome!

We might also look around what other languages do about this.

### PR DESCRIPTION
**Pull request by rofinn**
Closes #2

### CODE REVIEW

File: src/path.jl [hunk 1]
```diff
@@ -144,6 +132,39 @@ function parents(path::T) where {T <: AbstractPath}
...
+
+Join the path components into a new fulll path, equivalent to calling `joinpath`
+
+# Example
+
+julia> p"foo" / "bar"
+p"foo/bar"
+
+julia> p"foo" / "bar" / "baz"
+p"foo/bar/baz"
```
Comment (omus) [inline src/path.jl]: Just curious what happens if you do: `p"/foo" / "/bar"`

File: src/path.jl [hunk 2]
```diff
@@ -51,12 +51,8 @@ of the file containing the macro. Returns an empty Path if run from a REPL or
...
 """
 macro __PATH__()
-    if VERSION >= v"0.7-"
-        p = Path(dirname(string(__source__.file)))
-        return p===nothing ? :(Path()) : :($p)
-    else
-        return :(Path(@__DIR__()===nothing ? Path() : @__DIR__))
-    end
+    p = Path(dirname(string(__source__.file)))
+    return p===nothing ? :(Path()) : :($p)
```
Comment (omus) [inline src/path.jl]:
```suggestion
    return p === nothing ? :(Path()) : :($p)
```

File: src/path.jl [hunk 3]
```diff
@@ -67,12 +63,8 @@ containing the macro. Returns an empty Path if run from a REPL or if
...
 """
 macro __FILEPATH__()
-    if VERSION >= v"0.7-"
-        p = Path(string(__source__.file))
-        return p===nothing ? :(Path()) : :($p)
-    else
-        return :(Path(@__FILE__()===nothing ? Path() : @__FILE__))
-    end
+    p = Path(string(__source__.file))
+    return p===nothing ? :(Path()) : :($p)
```
Comment (omus) [inline src/path.jl]:
```suggestion
    return p === nothing ? :(Path()) : :($p)
```

File: src/path.jl (hunk 1 above)
Comment (rofinn) [reply src/path.jl]:
```
julia> p"/foo" / "/bar"
p"/bar"
```

Comment (rofinn) [reply src/path.jl]: That's the same behaviour as base.

```
julia> joinpath("/foo", "/bar")
"/bar"
```

Comment (omus) [reply src/path.jl]: Ok, thanks!

**Comment by rofinn** [comment]
Thanks for reviewing!

**Pull request merged**